core/src/main/java/org/apache/iceberg/rest/requests/CreateNamespaceRequest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
```

Comment:
Annotation classes are ignored if they are not present in the classpath. Relying on annotations brings in a new class of problems caused by the end user messing up their classpath, so I am strongly in favor of avoiding them. If Jackson can infer everything that we need it to without them, then we can use that inference. Otherwise I think we should make serialization more explicit.

I believe that we don't currently use Jackson annotations, so this change would also require updating all of our runtime Jars and their licenses.

Reply:
Let me check if Jackson can infer everything. I don't think it can - which is why I added these in."

I would be in favor of the more explicit SerDe as well given what's mentioned above (or using a library to generate the classes for us would be my preferred approach but I know that comes with its own set of issues).

Reply:
So I realized the issue. We can't validate the input if we just let Jackson infer things.

For example, if a server sends a `CreateNamespaceResponse` of `{ }`, it will not fail on deserialization and we will get `CreateNamespaceResponse(namespace=null, properties=null)`.

The annotations force it to go through the constructor, which is where we throw if the required / not-null field `namespace` isn't found.